src/main/kotlin/lotto/core/Lottos.kt

```diff
@@ -1,16 +1,8 @@
 package lotto.core
 
 class Lottos(private val lottos: List<Lotto>) : List<Lotto> by lottos {
-    fun countWinningRanks(winningNumbers: WinningNumbers): Map<WinningRank, Int> {
-        val results = this.map { checkWinningState(it, winningNumbers) }
-
-        return results.groupingBy { it }.eachCount()
-    }
-
-    private fun checkWinningState(
-        lotto: Lotto,
-        winningNumbers: WinningNumbers,
-    ): WinningRank {
-        return WinningRank.getWinningAmount(winningNumbers.countCommonNumbers(lotto))
+    fun countWinningRanks(winningLotto: WinningLotto): LottoResult {
+        val results = this.map { winningLotto.checkWinningState(it) }
+        return LottoResult(results.groupingBy { it }.eachCount())
```

Comment:
생성자 파라미터로 꽤 긴 로직이 들어가있어서, 코드가 잘 안 읽히는 것 같아요.
적절히 변수로 추출해서 조금 더 가독성을 높여주면 좋을 것 같습니다. 😃

```suggestion
        val winningRankCount = results.groupingBy { it }.eachCount()
        return LottoResult(winningRankCount)
```